packages/salesforcedx-vscode-core/src/index.ts

```diff
@@ -441,9 +443,12 @@ async function setupOrgBrowser(
 export async function activate(context: vscode.ExtensionContext) {
   const extensionHRStart = process.hrtime();
   // Telemetry
+  const isCliTelemetryAllowed = await isCLITelemetryAllowed(
```

Comment:
I don't think we need to add this check here, we can just have everything be part of the internals of the telemetry service.

Reply:
+1. I'm also wary of adding something like this to the activate function

Reply:
@xyc There is precedent for async behavior in activate(). What specifically is your concern?

Reply:
Performance primarily. This puts everything below on hold. Especially looks like it involves a CLI command. If this has to be in `activate` I would suggest to wrap in something like `Promise.all`, otherwise I would agree with what @lcampos suggests and read the value on demand.

Reply:
@rcoringrato-sfdc @xyc I want to clarify that my ask was not based on performance but about where the code belongs. Having the check be part of the params a consumer needs to provide did not make sense since that should be part of the internal checks the service runs. The service itself is checking for different environment variables, checking the CLI telemetry preference should be one of those and there is no value in having it be done outside of the service since it's not needed anywhere else.

Reply:
@lcampos Understood with regards to the placement of the check. My most recent update places the check in TelemetryService initialization.